src/algorithm/joint-configuration.hpp

```diff
@@ -0,0 +1,334 @@
+//
+// Copyright (c) 2016 CNRS
+//
+// This file is part of Pinocchio
+// Pinocchio is free software: you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation, either version
+// 3 of the License, or (at your option) any later version.
+//
+// Pinocchio is distributed in the hope that it will be
+// useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+// of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Lesser Public License for more details. You should have
+// received a copy of the GNU Lesser General Public License along with
+// Pinocchio If not, see
+// <http://www.gnu.org/licenses/>.
+
+#ifndef __se3_joint_configuration_hpp__
+#define __se3_joint_configuration_hpp__
+
+#include "pinocchio/multibody/visitor.hpp"
+#include "pinocchio/multibody/model.hpp"
+  
+namespace se3
+{
+
+  /**
+   * @brief      Integrate the model for a constant derivative during unit time
+   *
+   * @param[in]  model   Model that must be integrated
+   * @param[in]  data    Corresponding Data to the Model
+   * @param[in]  q       Initial configuration (size model.nq)
+   * @param[in]  v       Velocity (size model.nv)
+   * @param      result  Resulting configuration result (size model.nq)
+   */
+  inline void integrate(const Model & model,
+                             Data & data,
+                             const Eigen::VectorXd & q,
+                             const Eigen::VectorXd & v,
+                             Eigen::VectorXd & result);
+
+
+  /**
+   * @brief      Interpolate the model between two configurations
+   *
+   * @param[in]  model   Model to be interpolated
+   * @param[in]  data    Corresponding Data to the Model
+   * @param[in]  q1      Initial configuration vector (size model.nq)
+   * @param[in]  q2      Final configuration vector (size model.nq)
+   * @param[in]  u       u in [0;1] position along the interpolation.
+   * @param      result  The interpolated configuration (q1 if u = 0, q2 if u = 1)
+   */
+  inline void interpolate(const Model & model,
+                               Data & data,
+                               const Eigen::VectorXd & q1,
+                               const Eigen::VectorXd & q2,
+                               const double u,
+                               Eigen::VectorXd & result);
+
+
+  /**
+   * @brief      the constant derivative that must be integrated during unit time to go from q2 to q1
```

Comment:
Please change the API in such way that differentiate goes from q1 to q2 during one unit time to be consistent with the rest of the API.
Please use also q0 instead of q1 and q1 instead of q2.